src/Symfony/Component/Console/Messenger/ExecuteCommandHandler.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Messenger;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\BufferedOutput;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class ExecuteCommandHandler
+{
+    public function __construct(private readonly Application $application)
+    {
+    }
+
+    public function __invoke(ExecuteCommand $message): BufferedOutput
+    {
+        $input = new StringInput($message->input);
+        $output = new BufferedOutput();
+
+        $this->application->setCatchExceptions($message->catchExceptions);
```

Comment:
Should I clone the application here instead of having the service not-shared?

Reply:
I don't think I can do this as there's no way to access the current state of `Application::$catchExceptions`.